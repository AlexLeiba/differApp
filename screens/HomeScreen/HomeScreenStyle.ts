import React from "react";
import { StyleSheet } from "react-native";

export const stylesHomeScreen = StyleSheet.create({
  containerHomeScreen: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    alignItems: "center",
    height: "100%",
  },
});
