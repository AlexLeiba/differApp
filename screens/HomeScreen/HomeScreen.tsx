import react, { useEffect, useContext, useState } from "react";
import { useLogin } from "../../hooks/accounts/useLogin";
import { Text, SafeAreaView, View } from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { Button } from "../../components";
import { AuthContext } from "../../consts";
import { stylesHomeScreen } from "./HomeScreenStyle";
import jwtDecode from "jwt-decode";

export function HomeScreen() {
  const [loading, setLoading] = useState(true);
  const [dataToken, setDataToken] = useState();
  const [{ data }] = useLogin();

  useEffect(() => {
    setLoading(true);
    async function Fetch() {
      const token: any = await AsyncStorage.getItem("@token");
      const userData: any = jwtDecode(token);
      if (userData !== undefined) {
        setDataToken(userData);
      }
    }
    setLoading(false);
    Fetch();
  }, []);

  console.log(dataToken);

  const { signOut } = useContext(AuthContext);

  async function handleSignOut() {
    signOut();
    AsyncStorage.clear();
  }

  return (
    <SafeAreaView style={stylesHomeScreen.containerHomeScreen}>
      <Text style={{ fontSize: 20, marginTop: 50 }}>
        Welcome on the home page!
      </Text>
      <View style={{ marginBottom: 20 }}>
        <Button
          title="Log out"
          colorContainer="black"
          colorTitle="white"
          handlePress={handleSignOut}
          width={300}
          textAlign={"center"}
          borderRadius={4}
          padding={16}
        />
      </View>
    </SafeAreaView>
  );
}
