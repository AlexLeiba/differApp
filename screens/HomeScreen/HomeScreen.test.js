import { HomeScreen } from "./HomeScreen";
import { render } from "react-native-testing-library";

describe("HomePage component to not have changes", () => {
  it("should match snapshot", () => {
    const rendered = render(<HomeScreen />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
