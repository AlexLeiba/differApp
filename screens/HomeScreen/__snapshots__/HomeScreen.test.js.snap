// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component to not have changes should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginTop": 50,
      }
    }
  >
    Welcome on the home page!
  </Text>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 4,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 300,
        }
      }
      testID="undefined-button"
    >
      <Text
        colorTitle="white"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
              "textAlign": "center",
            },
          ]
        }
        testID="undefined-text"
        textAlign="center"
      >
        Log out
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
